csharp_dotnetcore/13.Basic-Bot-Template/Startup.cs

```diff
@@ -0,0 +1,217 @@
+        ///
+        /// For example, LUIS services are created here. This external services is configured
+        /// using the <see cref="BotConfiguration"/> class (based on the contents of your ".bot" file).
+        /// </summary>
+        /// <param name="config">The <see cref="BotConfiguration"/> object based on your ".bot" file.</param>
+        /// <returns>A <see cref="BotServices"/> representing client objects to access external services the bot uses.</returns>
+        /// <seealso cref="BotConfiguration"/>
+        /// <seealso cref="LuisRecognizer"/>
+        private static BotServices InitBotServices(BotConfiguration config)
+        {
+            var luisServices = new Dictionary<string, LuisRecognizer>();
+
+            foreach (var service in config.Services)
+            {
+                switch (service.Type)
+                {
+                    case ServiceTypes.Luis:
+                        {
+                            var luis = (LuisService)service;
+                            if (luis == null)
+                            {
+                                throw new InvalidOperationException("The LUIS service is not configured correctly in your '.bot' file.");
+                            }
+
+                            if (string.IsNullOrWhiteSpace(luis.AppId))
+                            {
+                                throw new InvalidOperationException("The LUIS Model Application Id ('appId') is required to run this sample. Please update your '.bot' file.");
+                            }
+
+                            if (string.IsNullOrWhiteSpace(luis.AuthoringKey))
+                            {
+                                throw new InvalidOperationException("The LUIS Authoring Key ('authoringKey') is required to run this sample. Please update your '.bot' file.");
+                            }
+
+                            if (string.IsNullOrWhiteSpace(luis.SubscriptionKey))
+                            {
+                                throw new InvalidOperationException("The Subscription Key ('subscriptionKey') is required to run this sample. Please update your '.bot' file.");
+                            }
+
+                            if (string.IsNullOrWhiteSpace(luis.Region))
+                            {
+                                throw new InvalidOperationException("The Region ('region') is required to run this sample. Please update your '.bot' file.");
+                            }
+
+                            var app = new LuisApplication(luis.AppId, luis.SubscriptionKey, luis.Region);
+                            var recognizer = new LuisRecognizer(app);
+                            luisServices.Add(luis.Name, recognizer);
+                            break;
+                        }
+                }
+            }
+
+            var connectedServices = new BotServices(luisServices);
+            return connectedServices;
+        }
+    }
```

Comment:
why isn't this in the bot?  This feels like bot stuff

Reply:
We want to cache the client and not re-create the connection for every turn..